src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -77,77 +77,48 @@ private static ExpressionSyntax GenerateNullLiteral()
-            {
-                return GenerateSingleLiteralExpression(type, (float)value, canUseFieldReference);
-            }
-            else if (value is double)
-            {
-                return GenerateDoubleLiteralExpression(type, (double)value, canUseFieldReference);
-            }
-            else if (value is decimal)
-            {
-                return GenerateLiteralExpression(type, (decimal)value, LiteralSpecialValues.DecimalSpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
-            }
-            else if (type == null || type.IsReferenceType || type.IsPointerType())
-            {
-                return GenerateNullLiteral();
-            }
-            else
-            {
-                return SyntaxFactory.DefaultExpression(type.GenerateTypeSyntax());
+                case bool val: return GenerateBooleanLiteralExpression(val);
+                case string val: return GenerateStringLiteralExpression(val);
+                case char val: return GenerateCharLiteralExpression(val);
+                case sbyte val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.SByteSpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v));
+                case short val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.Int16SpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v));
+                case int val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.Int32SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
+                case long val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.Int64SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
+                case byte val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.ByteSpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v));
+                case ushort val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.UInt16SpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, (uint)v));
+                case uint val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.UInt32SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
+                case ulong val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.UInt64SpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
+                case float val: return GenerateSingleLiteralExpression(type, val, canUseFieldReference);
+                case double val: return GenerateDoubleLiteralExpression(type, val, canUseFieldReference);
+                case decimal val: return GenerateLiteralExpression(type, val, LiteralSpecialValues.DecimalSpecialValues, null, canUseFieldReference, SyntaxFactory.Literal);
             }
+
+            return type == null || type.IsReferenceType || type.IsPointerType() || type.IsNullable()
```

Comment:
this line is different.  I added "type.isNullable()" here to address the reported bug.